### DISCUSSION TITLE
How to set maximum canvas size?

### DISCUSSION THREAD

**Discussion by cryo75** [CropperBlazor/Cropper.Blazor#236] [General]
I have the following cropper component set up:

![image](https://github.com/CropperBlazor/Cropper.Blazor/assets/3060986/1a5cd9f3-d833-4fb4-9d37-9e718ab87548)

The original image is very big.

How do I set the maximum width of the canvas?

And also, how can I enforce the crop box to stay within the bounds of the image? (If this cannot be set, then how do I set the canvas size to be the same size of the image?)

**Comment by MaxymGorn** [comment]
Hi @cryo75. You can experimenting with ViewMode for Cropper Component. See details instruction in this page: https://cropperblazor.github.io/examples/viewmodes#simple-usage
<img width="1279" alt="image" src="https://github.com/CropperBlazor/Cropper.Blazor/assets/50423072/ba2de35b-8ea8-44fa-89a3-3899737f3229">
Actually you can set up min and max dimensions in the following instruction with example of code: 
<img width="1274" alt="image" src="https://github.com/CropperBlazor/Cropper.Blazor/assets/50423072/b45bbaef-aeda-4f1e-a0e0-f121e785f78a">
Moreover you can see [code base](https://github.com/CropperBlazor/Cropper.Blazor/tree/dev/src/Cropper.Blazor/Client) of demo part site and find suitable combination: https://cropperblazor.github.io/demo
